source/simulator/core.cc

```diff
@@ -2101,7 +2108,7 @@ namespace aspect
             // Restore particles through stored copy of particle handler,
             // created in start_timestep(),
             // but only if this timestep is to be repeated.
-            if (particle_world.get() != nullptr)
+            for (const auto &particle_world : particle_worlds)
```

Comment:
This `const` is a bit misleading, because the function does modify the particle world. Only the unique_ptr is const. Since you dont use `const` in the other places, maybe drop it here as well.

```suggestion
            for (auto &particle_world : particle_worlds)
```